src/pages/profile/external-identities/FenceAccount.test.ts

```diff
@@ -0,0 +1,160 @@
+import { act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { h } from 'react-hyperscript-helpers';
+import { Ajax } from 'src/libs/ajax';
+import { authStore, FenceStatus } from 'src/libs/state';
+import { FenceAccount } from 'src/pages/profile/external-identities/FenceAccount';
+import { asMockedFn, renderWithAppContexts } from 'src/testing/test-utils';
+
+// Mocking for Nav.getLink
+jest.mock('src/libs/nav', () => ({
+  ...jest.requireActual('src/libs/nav'),
+  getLink: jest.fn(() => ''),
+  useRoute: jest.fn(() => 'fence-callback'),
+}));
+
+jest.mock('react-notifications-component', () => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+jest.mock('src/libs/ajax');
+
+const nhlbi = {
+  key: 'fence',
+  provider: {
+    key: 'fence',
+    name: 'NHLBI BioData Catalyst Framework Services',
+    expiresAfter: 30,
+    short: 'NHLBI',
+  },
+};
+
+const fenceStatus: FenceStatus = {
+  fence: {
+    issued_at: new Date(Date.now()),
+    username: 'bojackhorseman',
+  },
+};
+
+describe('FenceLink', () => {
+  describe('when the user has not linked a fence account', () => {
+    it('renders the login button', async () => {
+      // Arrange
+
+      asMockedFn(Ajax).mockImplementation(
+        () =>
+          ({
+            Metrics: { captureEvent: jest.fn() } as Partial<ReturnType<typeof Ajax>['Metrics']>,
+            User: { getFenceAuthUrl: jest.fn().mockReturnValue({ url: 'https://foo.bar' }) } as Partial<
+              ReturnType<typeof Ajax>['User']
+            >,
+          } as ReturnType<typeof Ajax>)
+      );
+      // Act
+      await act(async () => {
+        renderWithAppContexts(h(FenceAccount, nhlbi));
+      });
+
+      // Assert
+      expect(screen.getByText('Log in to NHLBI')).not.toBeNull();
```

Comment:
For what it's worth, it's sufficient to just call `getBy...`. `getBy...` queries throw an error if no matching element is found.